### PR TITLE
feat(dh): unify usage of em dash fallback pipe

### PR DESCRIPTION
**Pull request by dzhavat**
### DataHub

- unify usage of displaying an empty fallback value
- rename `emptyValue` pipe to `emDashFallback`
- update `yesNo` pipe to return on `null`/`undefined`
- adjust tests

## References

- https://github.com/Energinet-DataHub/geh-metering-point/issues/570

### CODE REVIEW

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html [hunk 1]
```diff
@@ -160,10 +136,8 @@
           {{ t('capacity') }}
         </dt>
         <dd class="watt-text-m">
-          <ng-container *ngIf="secondaryMasterData?.capacity; else emptyField">
-            {{ secondaryMasterData?.capacity }}
-            {{ t('units.kilowatt') }}
-          </ng-container>
+          {{ secondaryMasterData?.capacity | emDashFallback }}
```
Comment (frontend-specialisten) [inline libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html:139]: Question: wouldn't this change cause kilowatt to be shown even though capacity is not defined?

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html [hunk 2]
```diff
@@ -193,12 +165,8 @@
           {{ t('ratedCapacity') }}
         </dt>
         <dd class="watt-text-m">
-          <ng-container
-            *ngIf="secondaryMasterData?.ratedCapacity; else emptyField"
-          >
-            {{ secondaryMasterData?.ratedCapacity }}
-            {{ t('units.kilowatt') }}
-          </ng-container>
+          {{ secondaryMasterData?.ratedCapacity | emDashFallback }}
```
Comment (frontend-specialisten) [inline libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html:168]: Question: wouldn't this change cause kilowatt to be shown even though ratedCapacity is not defined?

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html [hunk 3]
```diff
@@ -212,12 +180,8 @@
           {{ t('ratedCurrent') }}
         </dt>
         <dd class="watt-text-m">
-          <ng-container
-            *ngIf="secondaryMasterData?.ratedCurrent; else emptyField"
-          >
-            {{ secondaryMasterData?.ratedCurrent }}
-            {{ t('units.ampere') }}
-          </ng-container>
+          {{ secondaryMasterData?.ratedCurrent | emDashFallback }}
```
Comment (frontend-specialisten) [inline libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html:183]: Question: wouldn't this change cause ampere to be shown even though ratedCurrent is not defined?

File: libs/dh/metering-point/feature-overview/src/lib/identity/dh-metering-point-identity.component.html [hunk 4]
```diff
@@ -19,28 +19,23 @@
...
   <h1 [title]="t('gsrnNumber')">{{ meteringPoint?.gsrnNumber }}</h1>
 
   <div class="watt-text-s">
     <span [title]="t('connectionState')">
-      <ng-container *ngIf="meteringPoint?.connectionState; else emptyField">
-        <dh-status-badge
-          [connectionState]="meteringPoint?.connectionState"
-        ></dh-status-badge>
-      </ng-container>
+      <dh-status-badge
```
Comment (frontend-specialisten) [inline libs/dh/metering-point/feature-overview/src/lib/identity/dh-metering-point-identity.component.html:36]: Question: before if connectionState wasn't defined a fallback value would be shown. Will this now going to be just blank and is that intentional?

**Review by frontend-specialisten — CHANGES_REQUESTED** [review]

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html (hunk 1 above)
Comment (dzhavat) [reply libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html:139]: Yes. good catch. One solution could be to make the `capacity` a param to the `units.kilowatt` translation. Another solution is to make a getter and return the display value from it or return `undefined` if `capacity` is `null`/`undefined`. A third solution is to also create a directive that displays the em dash.

Do you have a preference? I think I'll try the getter first.

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html (hunk 2 above)
Comment (dzhavat) [reply libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html:168]: Yes. good catch. I think the options are similar to my previous comment above (under `capacity`).

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html (hunk 3 above)
Comment (dzhavat) [reply libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html:183]: Yes. good catch. I think the options are similar to my previous comment above (under `capacity`).

File: libs/dh/metering-point/feature-overview/src/lib/identity/dh-metering-point-identity.component.html (hunk 4 above)
Comment (dzhavat) [reply libs/dh/metering-point/feature-overview/src/lib/identity/dh-metering-point-identity.component.html:36]: According to the `MeteringPointCimDto` interface, the `connectionState` property will always have a value. That was the reason for removing the fallback. From the code it loos like `connectionState` Input property might end up with `undefined` but that is mostly due to how the `meteringPoint` property is typed.

Considering all of this, do you think it's still relevant to display a fallback?

File: libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html (hunk 1 above)
Comment (dzhavat) [reply libs/dh/metering-point/feature-overview/src/lib/secondary-master-data/dh-secondary-master-data.component.html]: Went with the first option.

File: libs/dh/metering-point/feature-overview/src/lib/identity/dh-metering-point-identity.component.html (hunk 4 above)
Comment (frontend-specialisten) [reply libs/dh/metering-point/feature-overview/src/lib/identity/dh-metering-point-identity.component.html:36]: No I think it's totally fine :)

**Review by frontend-specialisten — APPROVED** [review]

**Pull request merged**